lib/data_hygiene/data_sync.rb

```diff
@@ -0,0 +1,152 @@
+
+      delete_topics_from_gov_delivery(to_be_deleted)
+
+      create_topics_on_gov_delivery(to_be_created)
+    end
+
+    def validate!
+      unless ENV["ALLOW_GOVDELIVERY_SYNC"] == "allow"
+        logger.info "Syncing GovDelivery has not been configured for this environment."
+        logger.info "Running this against production GovDelivery would be a really bad idea."
+        logger.info "If you're sure you want to run this, export ALLOW_GOVDELIVERY_SYNC='allow'"
+
+        abort
+      end
+
+      unless EmailAlertAPI.config.gov_delivery[:hostname] == "stage-api.govdelivery.com"
+        logger.info "It looks like you're running this sync with a non-staging GovDelivery configuration."
+        logger.info "Running this against production GovDelivery would be a really bad idea."
+        logger.info "If the GovDelivery staging hostname has changed, please update this application and try again."
+
+        abort
+      end
+    end
+
+    def update_delivery_id!
+      environment_account_code = EmailAlertAPI.config.gov_delivery[:account_code]
+
+      unless environment_account_code == PRODUCTION_ACCOUNT_CODE
+        logger.info "Updating topics in subscriber lists so that prefixes match account id for this environment..."
+
+        SubscriberList.update_all(
+          "gov_delivery_id = replace(gov_delivery_id, '#{PRODUCTION_ACCOUNT_CODE}_', '#{environment_account_code}_')"
+        )
+
+        logger.info "Done."
+      end
+    end
+
+    def subscriber_lists
+      @subscriber_lists ||= SubscriberList
+        .where.not(gov_delivery_id: nil)
+        .distinct
+        .pluck(:title, :gov_delivery_id)
+        .map do |title, gov_delivery_id|
+          [(title || "MISSING TITLE #{gov_delivery_id}").strip, gov_delivery_id]
+        end
+    end
+
+    def topics
+      @topics ||= Services.gov_delivery
+                    .fetch_topics["topics"]
+                    .map { |topic| [topic["name"], topic["code"]] }
+    end
+
+    def delete_topics_from_gov_delivery(list)
+      if list.blank?
+        logger.info "No topics to be deleted in GovDelivery."
+      else
+        logger.warn "Deleting remote topics.."
+
+        batch_size = [list.count, (list.count.to_f / THREAD_COUNT).ceil].min
+
+        threads = list.each_slice(batch_size).map do |batch|
+          Thread.new do
+            batch.each do |title, gov_delivery_id|
+              logger.warn "-- Deleting #{gov_delivery_id} - #{title}"
+              Services.gov_delivery.delete_topic(gov_delivery_id)
+            end
+          end
+        end
+        threads.each(&:join)
+
+        # GovDelivery delete topics asynchronously and are only eventually
+        # consistent on deletes. We wait here to give the deletes a chance to
+        # process, and then perform a retry if the create fails
+        logger.info 'Waiting for 30 seconds for topics to be asynchronously deleted'
+
+        @delete_wait.times do
```

Comment:
As discussed this is a hand-wavy thing as we know GovDelivery is eventual consistent, but have no real feel for timing.